diff --git a/core/lib/Drupal/Core/Config/TypedConfigManager.php b/core/lib/Drupal/Core/Config/TypedConfigManager.php
index c6ddfb5e9c..14d3a13b10 100644
--- a/core/lib/Drupal/Core/Config/TypedConfigManager.php
+++ b/core/lib/Drupal/Core/Config/TypedConfigManager.php
@@ -75,6 +75,13 @@ protected function getDiscovery() {
    */
   public function get($name) {
     $data = $this->configStorage->read($name);
+    if ($data === FALSE) {
+      // For a typed config the data MUST exist.
+      $data = [];
+      \Drupal::logger('config_sync')->error('Missing required data for typed configuration: %config', [
+        '%config' => $name,
+      ]);
+    }
     return $this->createFromNameAndData($name, $data);
   }
 
diff --git a/core/tests/Drupal/KernelTests/Config/TypedConfigTest.php b/core/tests/Drupal/KernelTests/Config/TypedConfigTest.php
index bc9c441d51..5a16b0ddcd 100644
--- a/core/tests/Drupal/KernelTests/Config/TypedConfigTest.php
+++ b/core/tests/Drupal/KernelTests/Config/TypedConfigTest.php
@@ -38,6 +38,10 @@ protected function setUp() {
   public function testTypedDataAPI() {
     /** @var \Drupal\Core\Config\TypedConfigManagerInterface $typed_config_manager */
     $typed_config_manager = \Drupal::service('config.typed');
+
+    // Test not existing data.
+    $typed_config = $typed_config_manager->get('config_test.not_existing');
+
     /** @var \Drupal\Core\Config\Schema\TypedConfigInterface $typed_config */
     $typed_config = $typed_config_manager->get('config_test.validation');
 
